plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1161,7 +1161,21 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       }
    }
 
+   void pause_at_block(block_num_type block_num) {
+      auto& chain = chain_plug->chain();
+
+      auto head = chain.head();
+      EOS_ASSERT(block_num > head.block_num(), invalid_pause_at_block_request,
+                 "Pause at block ${bn} <= chain head ${h}", ("bn", block_num)("h", head.block_num()));
+
+      fc_ilog(_log, "Set pause at block #${bn}", ("bn", block_num));
+      chain.set_pause_at_block_num(block_num);
+   }
+
    void resume() {
+      auto& chain = chain_plug->chain();
+      chain.set_pause_at_block_num(std::numeric_limits<block_num_type>::max());
```

Comment:
So a single `resume` will clear out any pause no matter how pause was initially triggered? Should a separate resume for pause_at_block_num?

Reply:
My thought was that the normal use case would be to use `pause_at_block` on non-BP nodes. I don't think there is a good use case for using it on a BP node. So I figured one `resume` would suffice. Yes, it resumes either type of pause, but that seems like an ok compromise to avoid adding an additional endpoint.